addons/binding/org.openhab.binding.konnected/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,131 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="konnected"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0 http://eclipse.org/smarthome/schemas/thing-description-1.0.0.xsd">
+	<!-- This is the Konnected Thing-Type -->
+	<thing-type id="module" extensible="switch,actuator,temperature,humidity">
+		<label>The Konnected Alarm Panel</label>
+		<description>The Konnected Module</description>
+		<channels>
+			<channel id="Zone_6" typeId="switch">
+				<label>Zone 6</label>
+				<description>Zone 6 Sensor</description>
+			</channel>
+			<channel id="Zone_7" typeId="actuator">
+				<label>The Out Pin</label>
+			</channel>
+		</channels>
+		<config-description-ref uri="thing-type:konnected:module" />
+	</thing-type>
+	<!-- Zone Channel Type -->
+	<channel-type id="switch">
+		<item-type>Contact</item-type>
+		<label>ON/OFF Switch</label>
+		<description>This zone is a read only switch type zone</description>
+		<state readOnly="true" />
+		<config-description>
+		<parameter name="channel_zone" type="text">
+                <label>Zone Number</label>
+                <description>The Zone Number of the channel.</description>
+                <default>6</default>
+                <options>
+                    <option value="1">1</option>
+                    <option value="2">2</option>
+                    <option value="3">3</option>
+                    <option value="4">4</option>
+                    <option value="5">5</option>
+                    <option value="6">6</option>
+                    <option value="7">Out</option>
+                </options>
+            </parameter>
+		</config-description>
+	</channel-type>
+	<channel-type id="temperature">
+		<item-type>Number:Temperature</item-type>
+		<label>Temperature</label>
+		<description>This zone measures temperature</description>
+		<state readOnly="true" />
+		<config-description>
```

Comment:
can you add to a config description that is referenced via a uri?   It didn't seem like to me when reading https://www.eclipse.org/smarthome/documentation/development/bindings/xml-reference.html that this was possible as it was an either or?  Seeing as how there are other parameters which differ for each channel having that one parameter in a config description wouldn't really change much as i would still need four separate config descriptions?  Unless it can be done using parameter groups in the config description somehow?

   <config-description>
      ...
    </config-description>
    OR
    <config-description-ref uri="{binding|thing-type|channel-type|any_other}:bindingID:..." />
  </thing-type>

Reply:
> there are other parameters which differ for each channel 

Sorry, I completely missed this fact (only saw the switch and the humidity channel, which both only had this single parameter. You are right that if there are different sets of parameters, it doesn't make much sense to extract the description, so just ignore my comment, please 🙄 .

Reply:
will do would be nice to be able to extract parameters like that though it would have made for a cleaner description in this use case and i am sure there are others as well.